### PR TITLE
MNT: Remove remaining 3.7 deprecations

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

Not sure why I missed listing these in #26865.

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by rcomer — DISMISSED** [review]
The PR number in the changenote filename is 2 out but otherwise this looks good to me.

File: lib/mpl_toolkits/axes_grid1/tests/test_axes_grid1.py [hunk 1]
```diff
@@ -432,13 +432,10 @@ def test_image_grid_single_bottom():
...
 
-def test_image_grid_label_mode_deprecation_warning():
-    imdata = np.arange(9).reshape((3, 3))
-
+def test_image_grid_label_mode_invalid():
     fig = plt.figure()
-    with pytest.warns(mpl.MatplotlibDeprecationWarning,
-                      match="Passing an undefined label_mode"):
-        grid = ImageGrid(fig, (0, 0, 1, 1), (2, 1), label_mode="foo")
+    with pytest.raises(mpl.ValueError, match="'foo' is not a valid value for mode"):
```
Comment (rcomer) [inline lib/mpl_toolkits/axes_grid1/tests/test_axes_grid1.py:437]:
```suggestion
    with pytest.raises(ValueError, match="'foo' is not a valid value for mode"):
```

Comment (QuLogic) [reply lib/mpl_toolkits/axes_grid1/tests/test_axes_grid1.py:437]: Oops, thought I had ran the tests before pushing, but obviously not.

**Comment by rcomer** [comment]
Looks like Appveyor is running out of time?  I re-started it but same result.

**Review by dstansby — APPROVED** [review]

**Pull request merged**